src/main/java/core/basesyntax/strategy/PurchaseOperationHandler.java

```diff
@@ -0,0 +1,12 @@
+package core.basesyntax.strategy;
+
+import core.basesyntax.dao.FruitStorageDao;
+import core.basesyntax.model.FruitTransaction;
+
+public class PurchaseOperationHandler implements OperationHandler {
+    @Override
+    public void updateFruitStorage(FruitTransaction fruitTransaction,
+                                   FruitStorageDao fruitStorageDao) {
+        fruitStorageDao.decreaseQuantity(fruitTransaction);
```

Comment:
Handling Purchase operation.
Check result balance before saving it in the Storage - it should be positive. Throw `RuntimeException` in case the balance is negative.
Currently, there is no check for a negative balance, which could lead to incorrect behavior if a purchase operation attempts to buy more fruits than are available in the storage.